src/main/java/org/commonjava/util/sidecar/services/ReportService.java

```diff
@@ -31,67 +34,107 @@
         this.trackedContent.appendUpload(upload);
+        logger.info(upload.toString());
     }
 
     public void appendDownload(TrackedContentEntry download){
         this.trackedContent.appendDownload(download);
+        logger.info(download.toString());
     }
 
+    public TrackedContent getTrackedContent()
+    {
+        return trackedContent;
+    }
 
-    @ConsumeEvent(value = FOLO_BUILD)
-    public void readReport(String path)
+    @ConsumeEvent(value = STARTUP_INIT)
+    public void loadReport(String path)
     {
-        HistoricalContentDTO content;
-        Path filePath = Path.of(  path , "" + getBuildConfigId() );
-        logger.info( "Loading build content history:" + filePath );
-        try
-        {
-            String json = Files.readString( filePath );
-            content = objectMapper.readValue( json, HistoricalContentDTO.class );
-            if ( content == null ){
-                logger.error( "Failed to read historical content which is empty." );
-            }
-            else {
-                for ( HistoricalEntryDTO download:content.getDownloads() )
-                {
-                     this.trackedContent.appendDownload( new TrackedContentEntry( new TrackingKey(getBuildConfigId()),
-                                                                                  download.getStoreKey(),
-                                                                                  AccessChannel.NATIVE,
-                                                                                  download.getOriginUrl(), download.getPath(), StoreEffect.DOWNLOAD, download.getSize(),
-                                                                                  download.getMd5(), download.getSha1(), download.getSha256() ));
+        if (getBuildConfigId() != null){
+            HistoricalContentDTO content;
+            Path filePath = Path.of(  path , File.separator, getBuildConfigId() );
+            logger.info( "Loading build content history:" + filePath );
+            try
+            {
+                String json = Files.readString( filePath );
+                content = objectMapper.readValue( json, HistoricalContentDTO.class );
+                if ( content == null ){
+                    logger.warn( "Failed to read historical content which is empty." );
+                }
+                else {
+                    for (HistoricalEntryDTO download:content.getDownloads()){
+                        this.historicalContentMap.put(download.getPath(),download);
+                    }
                 }
             }
+            catch ( IOException e)
+            {
+                logger.error( "convert file " + filePath + " to object failed" );
+            }
         }
-        catch ( IOException e)
-        {
-            logger.error( "convert file " + filePath + " to object failed" );
-        }
+    }
+
+    @ConsumeEvent(value = FOLO_BUILD)
+    public void logFoloDownload(String path)
```

Comment:
Seems this method is not called anywhere else. Should it be removed?

Reply:
It'll be triggered by event bus

Reply:
the trigger codes are FoloContentAccessResource.java line 91